package.json

```diff
@@ -17,30 +18,45 @@
     "test": "tests"
   },
   "scripts": {
-    "lint": "gulp lint",
-    "test": "gulp test && coffee tests/e2e/test.coffee",
-    "build": "gulp build",
-    "prepublish": "npm test && npm run build",
+    "test:unit": "mocha -r ts-node/register --reporter spec tests/**/*.spec.ts",
+    "test:e2e": "ts-node tests/e2e/test.ts",
+    "lint": "tslint lib/**/*.ts",
+    "test": "npm run lint && npm run build && npm run test:unit && npm run test:e2e",
+    "build": "rimraf ./build && npm run prettify && tsc && npm run readme",
+    "prepublish": "require-npm4-to-publish",
+    "prepublishOnly": "npm test",
+    "prettify": "prettier --use-tabs --single-quote --write lib/*.ts tests/**/*.ts",
```

Comment:
I'm not sure whether this is applicable in this case, but just as a heads up,  I needed to wrap my file patterns in single quotes to make them properly work [recursively ](https://github.com/prettier/prettier/issues/2078).